.github/workflows/build_containers.yml

```diff
@@ -0,0 +1,116 @@
+name: Creating container images
+run-name: Creating container images 🧉
+
+# Set the access for individual scopes, or use permissions: write-all
+permissions:
+  pull-requests: write
+  contents: write
+  packages: write
+
+on:
+  push:
+  pull_request:
+    types: [closed]
+    branches:
+      - main
+  workflow_dispatch:
+
+
+jobs:
```

Comment:
As a minor comment on the `jobs` section of the workflow: I used a [matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) to loop through repetitive args instead of writing them out twice ... i didn't run it on action, but if u faced any failed job, let me do some pushes to this branch

```yaml
jobs:
  setup-environment:
    runs-on: ubuntu-latest
    outputs:
      tag: ${{ steps.get_tag.outputs.tag }}
    permissions:
      contents: write
      packages: write
    steps:
      -
        name: Checkout repository
        uses: actions/checkout@v4
        with:
          fetch-depth: 0
      -
        name: Set up Python 3.X
        uses: actions/setup-python@v4
        with:
          python-version: "3.X"
  freva-with-dev-build-push:
    needs: setup-environment
    runs-on: ubuntu-latest
    strategy:
      matrix:
        image: [freva-dev, freva]
      -
        name: Checkout repository
        uses: actions/checkout@v4
        with:
          fetch-depth: 0
      -
        name: Set up QEMU
        uses: docker/setup-qemu-action@v1

      -
        name: Set up Docker Buildx
        uses: docker/setup-buildx-action@v1

      -
        name: Log in to the Container registry
        uses: docker/login-action@v2
        with:
          registry: ghcr.io
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}
      -
        name: Build and push freva image
        uses: docker/build-push-action@v4
        with:
          platforms: linux/amd64
          push: ${{ github.event_name == 'pull_request' && github.event.pull_request.merged == true && github.ref == 'refs/heads/main' }}
          file: .docker/Dockerfile
          build-args: |
            VERSION=${{ needs.setup-environment.outputs.tag }}
            binder=${{ matrix.image == 'freva-dev' && 'false' || 'true' }}
          tags: |
            ghcr.io/freva-clint/${{ matrix.image }}:${{ needs.setup-environment.outputs.tag }}
            ghcr.io/freva-clint/${{ matrix.image }}:latest

Reply:
That's a good point, I'll see if I can make a matrix job.